packages/dx-grid-core/src/plugins/sorting-state/reducers.ts

```diff
@@ -9,10 +9,10 @@ export const changeColumnSorting: PureReducer<ColumnSortingState, ChangeSortingP
 
   let nextSorting: any[] = [];
   if (keepOther === true) {
-    nextSorting = sorting.slice();
+    nextSorting = Array.prototype.slice.call(sorting);
   }
   if (Array.isArray(keepOther)) {
-    nextSorting = sorting.slice()
+    nextSorting = Array.prototype.slice.call(sorting)
```

Comment:
`slice` is not required because `filter` creates a copy.

Reply:
Without these changes, a demo will break down. You can see it if you remove my changes and run tests